gef.py

```diff
@@ -886,115 +915,135 @@ def __init__(self, elf: Elf, off: int) -> None:
+        EXCLUDE          = 0x80000000
+
+    sh_name      = 0
+    sh_type : Optional["Shdr.Type"] = None
+    sh_flags : Optional["Shdr.Flags"] = None
+    sh_addr      = 0
+    sh_offset    = 0
+    sh_size      = 0
+    sh_link      = 0
+    sh_info      = 0
+    sh_addralign = 0
+    sh_entsize   = 0
+
+    __name       = ""
 
     def __init__(self, elf: Optional[Elf], off: int) -> None:
         if elf is None:
             return
         elf.seek(off)
         endian = gef.arch.endianness
-        if elf.e_class == Elf.ELF_64_BITS:
-            self.sh_name, self.sh_type, self.sh_flags = struct.unpack(f"{endian}IIQ", elf.read(16))
-            self.sh_addr, self.sh_offset = struct.unpack(f"{endian}QQ", elf.read(16))
-            self.sh_size, self.sh_link, self.sh_info = struct.unpack(f"{endian}QII", elf.read(16))
-            self.sh_addralign, self.sh_entsize = struct.unpack(f"{endian}QQ", elf.read(16))
+        if elf.e_class == Elf.Class.ELF_64_BITS:
+            self.sh_name, sh_type, sh_flags = elf.read_and_unpack(f"{endian}IIQ")
+            self.sh_addr, self.sh_offset = elf.read_and_unpack(f"{endian}QQ")
+            self.sh_size, self.sh_link, self.sh_info = elf.read_and_unpack(f"{endian}QII")
+            self.sh_addralign, self.sh_entsize = elf.read_and_unpack(f"{endian}QQ")
         else:
-            self.sh_name, self.sh_type, self.sh_flags = struct.unpack(f"{endian}III", elf.read(12))
-            self.sh_addr, self.sh_offset = struct.unpack(f"{endian}II", elf.read(8))
-            self.sh_size, self.sh_link, self.sh_info = struct.unpack(f"{endian}III", elf.read(12))
-            self.sh_addralign, self.sh_entsize = struct.unpack(f"{endian}II", elf.read(8))
+            self.sh_name, sh_type, sh_flags = elf.read_and_unpack(f"{endian}III")
+            self.sh_addr, self.sh_offset = elf.read_and_unpack(f"{endian}II")
+            self.sh_size, self.sh_link, self.sh_info = elf.read_and_unpack(f"{endian}III")
+            self.sh_addralign, self.sh_entsize = elf.read_and_unpack(f"{endian}II")
 
+        self.sh_type = Shdr.Type(sh_type)
+        self.sh_flags = Shdr.Flags(sh_flags)
         stroff = elf.e_shoff + elf.e_shentsize * elf.e_shstrndx
 
-        if elf.e_class == Elf.ELF_64_BITS:
+        if elf.e_class == Elf.Class.ELF_64_BITS:
             elf.seek(stroff + 16 + 8)
-            offset = struct.unpack(f"{endian}Q", elf.read(8))[0]
+            offset = u64(elf.read(8))
         else:
             elf.seek(stroff + 12 + 4)
-            offset = struct.unpack(f"{endian}I", elf.read(4))[0]
+            offset = u32(elf.read(4))
         elf.seek(offset + self.sh_name)
-        self.sh_name = ""
+        self.__name = ""
         while True:
-            c = ord(elf.read(1))
+            c = u8(elf.read(1))
             if c == 0:
                 break
-            self.sh_name += chr(c)
+            self.__name += chr(c)
         return
 
+    @property
+    def name(self) -> str:
```

Comment:
Why do we need a `__name` at all?

Reply:
Fixed by d1bf612e78e056fb0325c7c48d6d49edca1ddaf3